docs/src/tutorials/deim_FitzHugh-Nagumo.md

```diff
@@ -0,0 +1,189 @@
+model via Petrov-Galerkin projection using the Proper Orthogonal Decomposition (POD) and 
+the Discrete Empirical Interpolation Method (DEIM). As an illustrative example, we consider 
+a nonlinear 1-D PDE from the realm of neuroscience.
+
+The FitzHugh-Nagumo system used in neuron modeling is a simplified version of the 
+Hodgkin-Huxley model, which describes in a detailed manner activation and deactivation 
+dynamics of a spiking neuron. The system is given as follows. For ``x\in [0,L], t\geq 0``,
+
+```math
+\begin{aligned}
+\varepsilon v_t(x,t)&=\varepsilon^2 v_{xx}(x,t)+f(v(x,t))-w(x,t)+c,\\
+w_t(x,t)&=bv(x,t)-\gamma w(x,t)+c,
+\end{aligned}
+```
+
+with nonlinear function ``f(v)=v(v-0.1)(1-v)``. The initial and boundary conditions are
+
+```math
+\begin{aligned}
+v(x,0)=0,\quad w(x,0)=0,\quad x\in [0,L],\\
+v_x(0,t)=-i_0(t),\quad v_x(L,t)=0,\quad t\geq 0,
+\end{aligned}
+```
+
+where the parameters are ``L=1``, ``\varepsilon=0.015``, ``b=0.5``, ``\gamma =2``, 
+``c=0.05``. The stimulus is ``i_0(t)=50000t^3\exp(-15t)``. The variables ``v`` and ``w`` 
+are voltage and recovery of voltage, respectively.
+
+In order to generate a POD-DEIM reduced-order model, we need to work through the following 
+steps:
+
+1. Collect data on full-order model trajectories and the nonlinear terms describing its evolution equation along the way.
+1. Based on the collected data, use POD to identify a low dimensional linear subspace of the system's state space that allows for embedding the full-order model's trajectories with minimal error.
+1. Project the model onto the identified subspace using DEIM to approximate nonlinear terms.
+
+For step 1, we first construct a 
+[`ModelingToolkit.PDESystem`](https://mtk.sciml.ai/stable/systems/PDESystem/) 
+describing the original FitzHugh-Nagumo model.
+
+```@example deim_FitzHugh_Nagumo
+using ModelingToolkit
+@variables x t v(..) w(..)
+Dx = Differential(x)
+Dxx = Dx^2
+Dt = Differential(t)
+const L = 1.0
+const ε = 0.015
+const b = 0.5
+const γ = 2.0
+const c = 0.05
+f(v) = v * (v - 0.1) * (1.0 - v)
+i₀(t) = 50000.0t^3 * exp(-15.0t)
+eqs = [ε * Dt(v(x, t)) ~ ε^2 * Dxx(v(x, t)) + f(v(x, t)) - w(x, t) + c,
+    Dt(w(x, t)) ~ b * v(x, t) - γ * w(x, t) + c]
+bcs = [v(x, 0.0) ~ 0.0,
+    w(x, 0) ~ 0.0,
+    Dx(v(0, t)) ~ -i₀(t),
+    Dx(v(L, t)) ~ 0.0]
+domains = [x ∈ (0.0, L),
+    t ∈ (0.0, 14.0)]
+ivs = [x, t]
+dvs = [v(x, t), w(x, t)]
+pde_sys = PDESystem(eqs, bcs, domains, ivs, dvs; name = Symbol("FitzHugh-Nagumo"))
+nothing # hide
+```
+
+Next, we apply finite difference discretization using 
+[MethodOfLines.jl](https://github.com/SciML/MethodOfLines.jl).
+
+```@example deim_FitzHugh_Nagumo
+using MethodOfLines
+N = 15 # equidistant discretization intervals
+dx = (L - 0.0) / N
+dxs = [x => dx]
+order = 2
+discretization = MOLFiniteDifference(dxs, t; approx_order = order)
+ode_sys, tspan = symbolic_discretize(pde_sys, discretization)
```

Comment:
@xtalax this is an interface break.

Reply:
Note that this might break in a future version, hopefully soon. Not your fault though @bowenszhu , we should make an issue to track it though.

Reply:
@YingboMa we should add (symbolic) time span to the system types so this can be supported better. It would also make modelingtoolkitize and reconstruction more automatic.